src/iop/censorize.c

```diff
@@ -0,0 +1,456 @@
+  }
+
+  output = temp;
+
+  // pixelate
+  if(pixel_radius != 0)
+  {
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(width, height, ch, input, output, pixel_radius, pixels_y, pixels_x) \
+  aligned(input, output:64) \
+  schedule(simd:static) collapse(2)
+#endif
+    for(size_t j = 0; j < pixels_y + 1; j++)
+      for(size_t i = 0; i < pixels_x + 1; i++)
+      {
+        // get the top left coordinate of the big pixel
+        const point_t tl = { CLAMP(2 * pixel_radius * i, 0, width - 1),
+                             CLAMP(2 * pixel_radius * j, 0, height - 1) };
+        // get the center of the big pixel
+        const point_t cc = { CLAMP(tl.x + pixel_radius, 0, width - 1),
+                             CLAMP(tl.y + pixel_radius, 0, height - 1) };
+        // get the bottom right coordinate of the big pixel
+        const point_t br = { CLAMP(cc.x + pixel_radius, 0, width - 1),
+                             CLAMP(cc.y + pixel_radius, 0, height - 1) };
+
+        // get the bounding box + center point coordinates
+        const point_t box[5] = { tl, { br.x, tl.y }, cc, { tl.x, br.y }, br };
+
+        // find the average color over the big pixel
+        float DT_ALIGNED_PIXEL RGB[4] = { 0.f };
+        for(size_t k = 0; k < 5; k++)
+        {
+          const float *const restrict pix_in = __builtin_assume_aligned(input + (width * box[k].y + box[k].x) * 4, 16);
+          for(size_t c = 0; c < 4; c++) RGB[c] += pix_in[c] / 5.f;
+        }
+
+        // paint the big pixel with solid color == average
+        for(size_t jj = tl.y; jj < br.y; jj++)
+          for(size_t ii = tl.x; ii < br.x; ii++)
+          {
+            float *const restrict pix_out = __builtin_assume_aligned(output + (jj * width + ii) * 4, 16);
+            for(size_t c = 0; c < 4; c++) pix_out[c] = RGB[c];
+          }
+      }
+
+    input = output;
+  }
+
+  // second blurring step
+  if(sigma_2 != 0.f)
+  {
+    output = out;
+
+    if(noise != 0.f)
+      make_noise(output, noise, width, height);
+
+    dt_gaussian_t *g = dt_gaussian_init(width, height, ch, RGBmax, RGBmin, sigma_2, 0);
+    if(!g) return;
+    dt_gaussian_blur_4c(g, input, output);
+    dt_gaussian_free(g);
+  }
+  else
+  {
+    output = out;
+    dt_simd_memcpy(input, output, width * height * 4);
+  }
+
+  if(noise != 0.f)
+    make_noise(output, noise, width, height);
+
+  if(piece->pipe->mask_display & DT_DEV_PIXELPIPE_DISPLAY_MASK)
+    dt_iop_alpha_copy(ivoid, ovoid, roi_out->width, roi_out->height);
+
+  dt_free_align(temp);
+}
+
+void gui_update(struct dt_iop_module_t *self)
+{
+  dt_iop_censorize_gui_data_t *g = (dt_iop_censorize_gui_data_t *)self->gui_data;
+  dt_iop_censorize_params_t *p = (dt_iop_censorize_params_t *)self->params;
+  dt_bauhaus_slider_set(g->radius_1, p->radius_1);
+  dt_bauhaus_slider_set(g->pixelate, p->pixelate);
+  dt_bauhaus_slider_set(g->radius_2, p->radius_2);
+  dt_bauhaus_slider_set(g->noise, p->noise);
+}
+
+/*
```

Comment:
Likewise, if not needed, better to remove.

Reply:
done